src/main/java/com/mael/ttt/players/HumanPlayer.java

```diff
@@ -23,24 +23,24 @@ public Mark getMark() {
         return mark;
     }
 
-    private String returnValidCellIndex() {
-        String cellIndex      = gameUI.getInput();
-        String validCellIndex = "[1-9]";
-
-        while (!cellIndex.matches(validCellIndex)) {
+    private String returnValidCellIndex(Board board) {
+        String cellIndex = gameUI.getInput();
+        while (isNotValidCellIndex(board, cellIndex)) {
             gameUI.printNotValidCellMessage();
             cellIndex = gameUI.getInput();
         }
-
         return cellIndex;
     }
 
+    private boolean isNotValidCellIndex(Board board, String cellIndex) {
+        return !gameUI.getBoardIndexesAsStrings(board.getSize()).contains(cellIndex);
+    }
+
     private String returnEmptyCellIndex(Board board, String cellIndex) {
```

Comment:
See comment above.
But also: Who knows and who should know whether a move is valid (i.e. not occupied and within the allowed range)?
Isn't this the Board itself?
Would it make sense to "move" the move validation to the board then?